### PR TITLE
Subset sum test well sampling strategy

### PR DESCRIPTION
**Pull request by MattsonCam**
In this pr, the subset sum sampling strategy is implemented to minimize the number of cells not removed from the train-validation set given the number of test wells specified for the minority group. Consequently, this strategy does not maximize the number of test wells in each group. Please let me know if you have any suggestions.

### CODE REVIEW

File: 1.train_models/utils/PlateTrainingSplits.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+class PlateTrainingSplits:
```
Comment (MikeLippincott) [inline 1.train_models/utils/PlateTrainingSplits.py:1]: Not sure if you wanted to add a docstring for this class?? Up to you, the class method doc strings look good

File: 1.train_models/utils/GreedySubsetSum.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        catdf = _welldf.groupby(_category_col)
+
+        # Determine the smalles reference category for sampling test wells
+        min_cat = plate_split.sum_train_val_category_cell_counts(
+            catdf,
+            _category_col,
+            _cell_count_col,
+            _test_well_count
+        )
```
Comment (MikeLippincott) [inline 1.train_models/utils/GreedySubsetSum.py]: Could this result possibly not be a reference category but rather the smallest category?

Comment (MikeLippincott) [reply 1.train_models/utils/GreedySubsetSum.py]: Unless reference in this case is a technical reference group not a biological reference group? If so, consider add a few words to the inline comments to make this more clear?

File: 1.train_models/utils/GreedySubsetSum.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import pandas as pd
+from PlateTrainingSplits import PlateTrainingSplits
+
+
+class GreedySubsetSum:
+
+    def __init__(self):
+        pass
+
+    def update_test_wells(self, _welldf, _category_col, _well_col, _cell_count_col, _test_well_count):
```
Comment (MikeLippincott) [inline 1.train_models/utils/GreedySubsetSum.py:13]: Consider adding a function to the docstring

**Review by MikeLippincott — APPROVED** [review]
LGTM. I made a few comments that might need addressed, up to you!

File: 1.train_models/utils/GreedySubsetSum.py (hunk 2 above)
Comment (MattsonCam) [reply 1.train_models/utils/GreedySubsetSum.py]: > Could this result possibly not be a reference category but rather the smallest category?

It should be both.

> Unless reference in this case is a technical reference group not a biological reference group? If so, consider add a few words to the inline comments to make this more clear?

The original intention was distinguish between biological groups to create a technical adjustment. Group are only constrained to specific columns in this implementation, so it's possible to also define these as technical groups.

File: 1.train_models/utils/GreedySubsetSum.py (hunk 3 above)
Comment (MattsonCam) [reply 1.train_models/utils/GreedySubsetSum.py:13]: I think you mean a docstring to the function. In the case of the init function, I could for completeness, but maybe not needed

File: 1.train_models/utils/PlateTrainingSplits.py (hunk 1 above)
Comment (MattsonCam) [reply 1.train_models/utils/PlateTrainingSplits.py:1]: Added

**Comment by MattsonCam** [comment]
Thanks @MikeLippincott, merging now!

**Pull request merged**